source/plugin/configuration/loaders.rst

```diff
@@ -102,23 +124,44 @@ it will be created. If it does exist, all contents will be overwritten.
     try {
         loader.save(rootNode);
     } catch(IOException e) {
-        // error
+        // handle error
     }
 
 Again, errors will be propagated as an ``IOException`` and must be handled.
 
-Example: Loading a default config from the plugin jar file
+.. tip::
+
+    It is recommended to save the config after loading it (for the first time after an update) to ensure that newly
+    added or migrated configuration options are written to disk. If you need to save the config afterwards it is
+    strongly recommended to do this outside of the main thread. See also common :doc:`/plugin/practices/bad` you should
+    avoid.
+
+Example: Loading a Default Config from the Plugin Jar File
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 .. code-block:: java
 
     import java.net.URL;
 
-    URL jarConfigFile = Sponge.getAssetManager().getAsset("defaultConfig.conf").get().getUrl();
-    ConfigurationLoader<CommentedConfigurationNode> loader =
-      HoconConfigurationLoader.builder().setURL(jarConfigFile).build();
+    rootNode = loader.load();
+    if (!rootNode.hasMapChildren()) { // is empty
+        this.logger.info("No config found - loading default");
+        URL jarConfigFile = Sponge.getAssetManager().getAsset("defaultConfig.conf").get().getUrl();
+        ConfigurationLoader<CommentedConfigurationNode> defaultLoader =
+                HoconConfigurationLoader.builder().setURL(jarConfigFile).build();
+        rootNode = defaultLoader.load();
+    }
 
 For this example it is important to note that the :javadoc:`AssetManager#getAsset(String)` method works relative to the
 plugin's asset folder. So, if in the above example the plugin ID is ``myplugin``, the ``defaultConfig.conf`` file
 must not lie in the jar file root, but instead in the directory ``assets/myplugin``. For more information, see
 :doc:`the Asset API page <../assets>`.
+
+.. note::
+    
+    If the config file cannot be found inside your plugin jar, then you will get a ``NoSuchElementException`` from the
+    ``Optional<Asset>.get()`` method. Please make sure that you configure your :doc:`build system </plugin/buildsystem>`
+    to include it in the jar.
+
+If you have an extra configuration class you can use a much easier approach, that also works if the only a part of your
```

Comment:
The comma should be after "configuration class", and not after "approach". Or replace "that" with "which".